dashboard/app/models/user.rb

```diff
@@ -1524,10 +1525,21 @@ def can_view_student_ai_chat_messages?
 
   # Students
   def has_ai_tutor_access?
-    !DCDO.get('ai-tutor-disabled', false) && (
-    permission?(UserPermission::AI_TUTOR_ACCESS) ||
-      (get_active_experiment_names_by_teachers.include?(AI_TUTOR_EXPERIMENT_NAME) &&
-      sections_as_student.any?(&:ai_tutor_enabled)))
+    return false if ai_tutor_access_denied || ai_tutor_feature_globally_disabled?
+    permission_for_ai_tutor? || in_ai_tutor_experiment_with_enabled_section?
+  end
+
+  private def ai_tutor_feature_globally_disabled?
+    DCDO.get('ai-tutor-disabled', false)
```

Comment:
Consider highlighting new DCDO flag with a note in the PR summary

Reply:
Not new, refactored: https://github.com/code-dot-org/code-dot-org/pull/57073/files#diff-0a5e79b422e8d2a2b8e5d9cc2203f62899ee74b65da749306b3d5f7c8b21a2d2L1527

But it is a good call to mention the refactor in case we see unrelated breakages.